File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/exportribbonxml/cs/exportribbonxml1.cs
// <snippetexportribbonxml1>


 // Connect to the Organization service. 
 // The using statement assures that the service proxy will be properly disposed.
 using (_serviceProxy = new OrganizationServiceProxy(serverConfig.OrganizationUri, serverConfig.HomeRealmUri,serverConfig.Credentials, serverConfig.DeviceCredentials))
 {
  // This statement is required to enable early-bound type support.                  
  _serviceProxy.EnableProxyTypes();
  
  //Create export folder for ribbon xml files if not already exist.
  if (!Directory.Exists(exportFolder))
      Directory.CreateDirectory(exportFolder);

  //Retrieve the Application Ribbon
  RetrieveApplicationRibbonRequest appribReq = new RetrieveApplicationRibbonRequest();
  RetrieveApplicationRibbonResponse appribResp = (RetrieveApplicationRibbonResponse)_serviceProxy.Execute(appribReq);

  System.String applicationRibbonPath = Path.GetFullPath(exportFolder + "\\applicationRibbon.xml");
  File.WriteAllBytes(applicationRibbonPath, unzipRibbon(appribResp.CompressedApplicationRibbonXml));
  //Write the path where the file has been saved.
  Console.WriteLine(applicationRibbonPath);
  //Retrieve system Entity Ribbons
  RetrieveEntityRibbonRequest entRibReq = new RetrieveEntityRibbonRequest() { RibbonLocationFilter = RibbonLocationFilters.All };

  foreach (System.String entityName in entitiesWithRibbons)
  {
   entRibReq.EntityName = entityName;
   RetrieveEntityRibbonResponse entRibResp = (RetrieveEntityRibbonResponse)_serviceProxy.Execute(entRibReq);

   System.String entityRibbonPath = Path.GetFullPath(exportFolder + "\\" + entityName + "Ribbon.xml");
   File.WriteAllBytes(entityRibbonPath, unzipRibbon(entRibResp.CompressedEntityXml));
   //Write the path where the file has been saved.
   Console.WriteLine(entityRibbonPath);
  }

  //Check for custom entities
  RetrieveAllEntitiesRequest raer = new RetrieveAllEntitiesRequest() { EntityFilters = EntityFilters.Entity };

  RetrieveAllEntitiesResponse resp = (RetrieveAllEntitiesResponse)_serviceProxy.Execute(raer);

  foreach (EntityMetadata em in resp.EntityMetadata)
  {
   if (em.IsCustomEntity == true && em.IsIntersect == false)
   {
    entRibReq.EntityName = em.LogicalName;
    RetrieveEntityRibbonResponse entRibResp = (RetrieveEntityRibbonResponse)_serviceProxy.Execute(entRibReq);

    System.String entityRibbonPath = Path.GetFullPath(exportFolder + "\\" + em.LogicalName + "Ribbon.xml");
    File.WriteAllBytes(entityRibbonPath, unzipRibbon(entRibResp.CompressedEntityXml));
    //Write the path where the file has been saved.
    Console.WriteLine(entityRibbonPath);
   }
  }
 }

// </snippetexportribbonxml1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/exportribbonxmlvb/vb/exportribbonxml1.vb
' <snippetexportribbonxml1>


 ' Connect to the Organization service. 
 ' The using statement assures that the service proxy will be properly disposed.
             _serviceProxy = ServerConnection.GetOrganizationProxy(serverConfig)
 Using _serviceProxy
  ' This statement is required to enable early-bound type support.                  
  _serviceProxy.EnableProxyTypes()

  'Create export folder for ribbon xml files if not already exist.
  If Not Directory.Exists(exportFolder) Then
      Directory.CreateDirectory(exportFolder)
  End If

  'Retrieve the Application Ribbon
  Dim appribReq As New RetrieveApplicationRibbonRequest()
  Dim appribResp As RetrieveApplicationRibbonResponse = CType(_serviceProxy.Execute(appribReq), RetrieveApplicationRibbonResponse)

  Dim applicationRibbonPath As String = Path.GetFullPath(exportFolder &amp; "\applicationRibbon.xml")
  File.WriteAllBytes(applicationRibbonPath, unzipRibbon(appribResp.CompressedApplicationRibbonXml))
  'Write the path where the file has been saved.
  Console.WriteLine(applicationRibbonPath)
  'Retrieve system Entity Ribbons
  Dim entRibReq As New RetrieveEntityRibbonRequest() With {.RibbonLocationFilter = RibbonLocationFilters.All}

  For Each entityName As String In entitiesWithRibbons
   entRibReq.EntityName = entityName
   Dim entRibResp As RetrieveEntityRibbonResponse = CType(_serviceProxy.Execute(entRibReq), RetrieveEntityRibbonResponse)

   Dim entityRibbonPath As String = Path.GetFullPath(exportFolder &amp; "\" &amp; entityName &amp; "Ribbon.xml")
   File.WriteAllBytes(entityRibbonPath, unzipRibbon(entRibResp.CompressedEntityXml))
   'Write the path where the file has been saved.
   Console.WriteLine(entityRibbonPath)
  Next entityName

  'Check for custom entities
  Dim raer As New RetrieveAllEntitiesRequest() With {.EntityFilters = EntityFilters.Entity}

  Dim resp As RetrieveAllEntitiesResponse = CType(_serviceProxy.Execute(raer), RetrieveAllEntitiesResponse)

  For Each em As EntityMetadata In resp.EntityMetadata
   If em.IsCustomEntity = True AndAlso em.IsIntersect = False Then
    entRibReq.EntityName = em.LogicalName
    Dim entRibResp As RetrieveEntityRibbonResponse = CType(_serviceProxy.Execute(entRibReq), RetrieveEntityRibbonResponse)

    Dim entityRibbonPath As String = Path.GetFullPath(exportFolder &amp; "\" &amp; em.LogicalName &amp; "Ribbon.xml")
    File.WriteAllBytes(entityRibbonPath, unzipRibbon(entRibResp.CompressedEntityXml))
    'Write the path where the file has been saved.
    Console.WriteLine(entityRibbonPath)
   End If
  Next em
 End Using

' </snippetexportribbonxml1>